### PR TITLE
[WIP] Setup tests for contracts

### PR DESCRIPTION
**Pull request by wilsonplau**
- Use `smock` to mock `KetlAttestationToken` to allow for testing
- Basic setup for contracts for testing
- Add basic test for KetlCred

### CODE REVIEW

File: test/OBSSStorage.KetlKred.test.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      obssStorageContract,
+      ketlCredContract,
+    } = await loadFixture(setupOBSS)
+    await feedsContract.addFeed(MOCK_CID)
+    expect(await feedsContract.lastFeedId()).to.equal(1)
+    await obssStorageContract.addFeedPost({
+      feedId: 0,
+      postMetadata: MOCK_CID,
+    })
+    await obssStorageContract.connect(user1).addFeedReaction({
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:106]: Let's also test switching reaction from from 1 to 2 (or from 0 to 1, can't quite remember):
User has `reactionType = 1`, ketl sends `addFeedReaction({ reactionType: 2 })`, it shouldn't throw an error, it should switch reactions from 1 to 2

File: test/OBSSStorage.KetlKred.test.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      }
+    )) as KetlCred
+    await ketlCredContract.deployed()
+
+    const feedsContractFactory = await ethers.getContractFactory('Feeds')
+    const feedsContract = (await upgrades.deployProxy(feedsContractFactory, [
+      ketlAttestationContractMock.address,
+      0,
+      deployer.address,
+    ])) as Feeds
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:57]: Can we avoid `as` here? Can we do `const feedsContract: Feeds = ...`

File: test/OBSSStorage.KetlKred.test.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        zeroAddress,
+        version,
+        ketlCredContract.address,
+        profilesContract.address,
+        feedsContract.address,
+      ],
+      {
+        initializer: 'initialize',
+      }
+    )) as OBSSStorage
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:73]: Maybe we can do it without `as`?

File: test/OBSSStorage.KetlKred.test.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    await profilesContract.deployed()
+
+    const ketlCredContractFactory = await ethers.getContractFactory('KetlCred')
+    const ketlCredContract = (await upgrades.deployProxy(
+      ketlCredContractFactory,
+      ['Ketl', 'KETL', 0, deployer.address],
+      {
+        initializer: 'initializeKetlCred',
+      }
+    )) as KetlCred
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:49]: Maybe we can do it without `as`?

File: test/OBSSStorage.KetlKred.test.ts (hunk 1 above)
Comment (wilsonplau) [reply test/OBSSStorage.KetlKred.test.ts:106]: I'll definitely build out a full suite of tests for thsi!

File: test/OBSSStorage.KetlKred.test.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+    const profilesContract = (await upgrades.deployProxy(
+      profilesContractFactory,
+      [ketlAttestationContractMock.address, 0, deployer.address]
+    )) as Profiles
+    await profilesContract.deployed()
+
+    const ketlCredContractFactory = await ethers.getContractFactory('KetlCred')
+    const ketlCredContract = (await upgrades.deployProxy(
+      ketlCredContractFactory,
+      ['Ketl', 'KETL', 0, deployer.address],
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:45]: As I remember, we used `'kekl', 'KEK'` for testing, maybe we should use it here as well

File: test/OBSSStorage.KetlKred.test.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+      },
+    ])
+    ketlAttestationContractMock.balanceOf.returns(1)
+    ketlAttestationContractMock.currentTokenId.returns(1)
+
+    const profilesContractFactory = await ethers.getContractFactory('Profiles')
+    const profilesContract = (await upgrades.deployProxy(
+      profilesContractFactory,
+      [ketlAttestationContractMock.address, 0, deployer.address]
+    )) as Profiles
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:39]: Maybe we can do it without `as`?

Comment (T-Damer) [reply test/OBSSStorage.KetlKred.test.ts:39]: Check my comment below about this

File: test/OBSSStorage.KetlKred.test.ts [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+import { expect } from 'chai'
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers'
+import { smock } from '@defi-wonderland/smock'
+import { version } from '../package.json'
+
+describe('OBSSStorage: KetlKred', () => {
+  async function setupOBSS() {
+    const [deployer, user1, user2, user3] = await ethers.getSigners()
+
+    const ketlAttestationContractMock = await smock.fake([
```
Comment (T-Damer) [inline test/OBSSStorage.KetlKred.test.ts:13]: Should we generate it instead of writing it by hand?

**Comment by wilsonplau** [comment]
Closed in favor of https://github.com/BigWhaleLabs/obss-storage-contract/pull/38.

**Pull request closed without merging**